<launch>

    <arg name="algorithm" default="Astar"/>
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="orientation_pr" default ="15"/>
    <arg name="grid_pr" default="0.5"/>
    <arg name="x_pos" default="-4.0"/>
    <arg name="y_pos" default="-3.1"/>
    <arg name="z_pos" default="0.0"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find motion_planning)/world/maze.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <param name="robot_description" command="$(find xacro)/xacro '$(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro'" />
    <param name="obstacles_file" value="$(find motion_planning)/config/obstacles.yaml"/>
    <param name="orientation_pr" value="$(arg orientation_pr)"/>
    <param name="grid_pr" value="$(arg grid_pr)"/>

    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

</launch>